documentation/asciidoc/topics/proc_enabling_client_cert_authn.adoc

```diff
@@ -0,0 +1,38 @@
+[id='enabling-client-certificate-authentication_{context}']
+= Enabling client certificate authentication
+
+[role="_abstract"]
+{brandname} clusters do not use client certificate authentication by default.
+Complete the following procedure to enable it.
+
+You can configure {brandname} to verify client identities from certificates in a trust store in two ways:
+
+`Validate`:: Requires a trust store that contains only the signing certificate, which is typically a Certificate Authority (CA). Any client that presents a certificate signed by the CA can connect to {brandname}.
+`Authenticate`:: Requires a trust store that contains all client certificates in addition to the signing certificate. Only clients that present a signed certificate that is present in the trust store can connect to {brandname}.
+
+.Procedure
+
+. Set either `Validate` or `Authenticate` as the value for the `spec.security.endpointEncryption.clientCert` field in your `Infinispan` CR.
++
+[NOTE]
+====
+The default value is `None`.
+====
++
+. Optionally specify the secret that contains the client trust store with the `spec.security.endpointEncryption.clientCertSecretName` field.
++
+By default {ispn_operator} uses a trust store secret named `{example_crd_name}-client-cert-secret`.
+If you want to use this secret you do not need to specify the `clientCertSecretName` field.
+However, you must create the secret if it does not already exist.
++
```

Comment:
> I don't know exact place, but we need to add notes, that operator will set owner reference for the provided secrets and will garbage collected them when Infinispan CR will be deleted, Also is not possible to share Secrets between instances.
> @ryanemerson wdyt about this?

Agreed @dmvolod.

We should maybe add this here as a `[NOTE]` @oraNod.

Something along the lines of:

"In order to support the updating of truststore secret values at runtime, it's necessary for the operator to add a owner reference with `controller: true` to the truststore Secret. Consequently, it's not possible for truststore Secrets to be utilised by multiple Infinispan CRs. Furhermore, upon Infinispan CR deletion the configured Secret will also be deleted."

[GC](https://kubernetes.io/docs/concepts/workloads/controllers/garbage-collection/) docs if it helps to understand what's going on.

This also applies to the TLS keystore secrets, so we need to update the docs in that section as well.

Reply:
@ryanemerson when you say TLS keystore secrets are you referring to `spec.security.endpointEncryption.certSecretName`?

Reply:
Bingo!

Reply:
thanks @ryanemerson and @dmvolod I added a note to two spots in the doc. hopefully the changes look good.